x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts

```diff
@@ -613,6 +614,7 @@ export const formatAboutStepData = (
     timestamp_override: timestampOverride !== '' ? timestampOverride : undefined,
     timestamp_override_fallback_disabled: timestampOverrideFallbackDisabled,
     ...(!isEmpty(note) ? { note } : {}),
+    max_signals: maxSignals ? maxSignals : undefined,
```

Comment:
Setting it like this causes the behaviour that @ARWNightingale described [in this comment](https://github.com/elastic/kibana/issues/173593#issuecomment-2092591834), where the form accepts `0`, but then gets replaced with the default value.

Replacing this with:
```
max_signals: maxSignals ?? undefined
```
will accepts `0` as a value, and the validation will fail on the API request, which I think is better UX.

Reply:
Replaced it with:
```
max_signals: Number.isSafeInteger(maxSignals) ? maxSignals : undefined,
```